### PR TITLE
Possible PR for #3639, KeyError on Preview Link

### PR DESCRIPTION
**Pull request by pkamp3**
One possible solution: If the session key does not exist when visiting a `/preview`, fall back to the "preview error page" (there will be a dialog).  The user can then try previewing the article again.

The most common reasons for the missing key:
- The user bookmarked the `/preview` link and is visiting it in a new session
- A user pastes a link to another user via email or IM

### CODE REVIEW

**Comment by pkamp3** [comment]
Discussion for #3639 can [be found here](https://github.com/wagtail/wagtail/issues/3639).

File: wagtail/wagtailadmin/tests/test_pages_views.py [hunk 1]
```diff
@@ -1550,6 +1550,27 @@ def test_preview_on_edit(self):
...
+        response = self.client.get(preview_url)
+
+        # Check the HTML response
+        self.assertEqual(response.status_code, 200)
+
+        # We should have an error page because we are unable to
+        # preview; the page key was not in the session.
+        self.assertContains(
+            response,
+            "title>Wagtail - Preview error</title>"
```
Comment (BertrandBordage) [inline wagtail/wagtailadmin/tests/test_pages_views.py]: `<title>` is missing one character.
Also, add the argument `html=True` to `assertContains` when comparing HTML. See [the Django docs about it](https://docs.djangoproject.com/en/1.11/topics/testing/tools/#django.test.SimpleTestCase.assertContains).

File: wagtail/wagtailadmin/tests/test_pages_views.py [hunk 2]
```diff
@@ -1550,6 +1550,27 @@ def test_preview_on_edit(self):
...
+        # get() without corresponding post(), key not set.
+        response = self.client.get(preview_url)
+
+        # Check the HTML response
+        self.assertEqual(response.status_code, 200)
+
+        # We should have an error page because we are unable to
+        # preview; the page key was not in the session.
+        self.assertContains(
+            response,
```
Comment (BertrandBordage) [inline wagtail/wagtailadmin/tests/test_pages_views.py]: I’m surprised that it’s working using the `Response` object. Please use `response.content` instead.

File: wagtail/wagtailadmin/tests/test_pages_views.py [hunk 3]
```diff
@@ -1550,6 +1550,27 @@ def test_preview_on_edit(self):
...
+            "title>Wagtail - Preview error</title>"
+        )
+        self.assertContains(
+            response,
+            "<h1>Preview error</h1>"
+        )
```
Comment (BertrandBordage) [inline wagtail/wagtailadmin/tests/test_pages_views.py]: Same comments as before.

File: wagtail/wagtailadmin/tests/test_pages_views.py (hunk 2 above)
Comment (gasman) [inline wagtail/wagtailadmin/tests/test_pages_views.py]: This is correct - `assertContains` specifically works with Response objects, as per https://docs.djangoproject.com/en/1.11/topics/testing/tools/#django.test.SimpleTestCase.assertContains. (Perhaps you're thinking of assertIn?)

Comment (pkamp3) [inline wagtail/wagtailadmin/tests/test_pages_views.py]: It looks like `assertContains` has some sugar for us behind the scenes:

https://github.com/django/django/blob/master/django/test/testcases.py#L352-L369

Would we prefer an  `assertHTMLEqual` (for both) instead?

Comment (BertrandBordage) [inline wagtail/wagtailadmin/tests/test_pages_views.py]: That’s right, sorry about that ;)

**Pull request merged**